fluid/PaddleCV/image_classification/eval_int8.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+import os
+import numpy as np
+import time
+import sys
+import paddle
+import paddle.fluid as fluid
+import models
+import reader
+import argparse
+import functools
+from models.learning_rate import cosine_decay
+from utility import add_arguments, print_arguments
+import paddle.fluid.profiler as profiler
+
+parser = argparse.ArgumentParser(description=__doc__)
+add_arg = functools.partial(add_arguments, argparser=parser)
+# yapf: disable
+add_arg('batch_size',       int,  256,                 "Minibatch size.")
+add_arg('use_gpu',          bool, True,                "Whether to use GPU or not.")
+add_arg('class_dim',        int,  1000,                "Class number.")
+add_arg('image_shape',      str,  "3,224,224",         "Input image size")
+add_arg('with_mem_opt',     bool, True,                "Whether to use memory optimization or not.")
+add_arg('pretrained_model', str,  None,                "Whether to use pretrained model.")
+add_arg('skip_batch_num',   int,  5,                    "Skip batch num.")
+add_arg('use_transpiler',   bool, True,                 'Whether to use transpiler.')
+add_arg('use_fake_data',    bool, False,                'If set, use fake data instead of real data.')
+add_arg('iterations',	    int,  100,               	'Fake data iterations')
+add_arg('profiler',	    bool, False,                'If true, do profiling.')
+# yapf: enable
+
+model_list = [m for m in dir(models) if "__" not in m]
+
+
+def user_data_reader(data):
+    '''
+    Creates a data reader for user data.
+    '''
+
+    def data_reader():
+        while True:
+            for b in data:
+                yield b
+
+    return data_reader
+
+
+def eval(args):
+    # parameters from arguments
+    class_dim = args.class_dim
+    pretrained_model = args.pretrained_model
+    with_memory_optimization = args.with_mem_opt
+    skip_batch_num = args.skip_batch_num
+    if skip_batch_num >= args.iterations:
+       print("Please ensure the skip_batch_num less than iterations.")
+       sys.exit(0)
+    image_shape = [int(m) for m in args.image_shape.split(",")]
+
+
+    if with_memory_optimization:
+        fluid.memory_optimize(fluid.default_main_program())
+
+    place = fluid.CUDAPlace(0) if args.use_gpu else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+
+    [infer_program, feed_dict,
+     fetch_targets] = fluid.io.load_inference_model(pretrained_model, exe)
+    
+    program = infer_program.clone()
+    if args.use_transpiler:
+        inference_transpiler_program = program.clone()
+        t = fluid.transpiler.InferenceTranspiler()
+        t.transpile(inference_transpiler_program, place)
+        program = inference_transpiler_program
+
+    fake_data = [(
+        np.random.rand(image_shape[0] * image_shape[1] * image_shape[2]).astype(np.float32),
+        np.random.randint(1, class_dim)) for _ in range(1)]
```

Comment:
why loop with `range(1)`?
Should this be put in `if use_fake_data`?

Reply:
it's just generated the random data for label.